server.go

```diff
@@ -553,6 +553,8 @@ func (s *Server) receiveMessage(m Message) error {
 		s.holder.recalculateCaches()
 	case *NodeEvent:
 		s.cluster.ReceiveEvent(obj)
+	case *Instruction:
+		s.handleInstruction(obj)
```

Comment:
I'm not sure I understand the need for this generic instruction type. To me, it feels like it will hide internal communication complexity instead of making it obvious in our proto definition. I'm not necessarily opposed to it... just want to understand why you chose to go this route.

Reply:
I don't really have a good argument for using `Instruction` over having a specific message type for each instruction (and there's currently only one instruction type anyway). I can change it to be specific to the instruction type.